src/main/java/com/client/core/base/util/TriggerUtil.java

```diff
@@ -42,13 +46,22 @@ public static boolean isReturnValue(String key) {
 	public static <E extends BullhornEntity> E populateEntity(Integer entityID, Class<E> type, Map<String, Object> values, Supplier<E> constructor,
 															  Set<String> fields) {
 		E entity = Optional.of(entityID)
-				.filter(id -> id != null && id > 0)
+				.filter(id -> id > 0)
 				.map(id -> getBullhornData().findEntity(type, id, fields))
 				.orElseGet(constructor);
 
 		values.entrySet().forEach( entry -> {
 			try {
-				getEntityChanger().setField(entity, entry.getKey(), entry.getValue());
+				if (entry.getValue() instanceof Map<?,?> && ((Map<?, ?>) entry.getValue()).containsKey("replaceAll") && AssociationEntity.class.isAssignableFrom(type)) {
+                    AssociationField<? extends AssociationEntity, ? extends BullhornEntity> associationField = AssociationFactory.getAssociationField((Class<? extends AssociationEntity>) type, entry.getKey());
+                    getEntityChanger().setField(entity, entry.getKey(),
+							convertReplaceAllToEntityOneToMany((Map<String, List<Integer>>) entry.getValue(),
+									associationField,
+									Utility.extractAssociationFieldsFromParent(fields, associationField.getAssociationFieldName()))
+					);
```

Comment:
In an effort to keep `populateEntity` as simple and readable as possible, can we extract this code into its own function? Maybe even two?

Something like:
```suggestion
    if (containsReplaceAll(entry, type)) {
        AbstractEntity value = convertToOneToMany(entry, type, fields);
	getEntityChanger().setField(entity, entry.getKey(),);
    }
```